### ISSUE TITLE
Add state=None kwarg to all normalisation layers?

### ISSUE THREAD

**Issue by jatentaki** [patrick-kidger/equinox#336]
The kind of normalisation layer is often considered a hyperparameter so it would be convenient if they all had the same `__call__` interface. I am therefore suggesting adding a dummy `state=` kwarg to normalisation layers other than `BatchNorm` such that one can easily build generic architectures with `__call__(self, x, state, key) -> tuple[Any, State]` API, where the state for stateless operations is simply `None`. This would be similar to how all layers take the `key` argument even if they are not rng-based. I am specifically interested in patching `eqxvision` to support the latest `equinox` release.

**Comment by patrick-kidger** [comment]
Yup, that sounds good to me! I'd be happy to take a PR on that.

**Comment by patrick-kidger** [comment]
What you've written sounds good to me! I think that's the only issue that needs taking care of.

Note that Equinox has a built-in (internal use only) sentinel value [here](https://github.com/patrick-kidger/equinox/blob/a514a6f4f207d712e6c01d5192e6e057eee09c9c/equinox/_custom_types.py#L9), which has a nice repr when displayed in the docs.

**Comment by patrick-kidger** [comment]
I've gone ahead and implemented this in #339.

Let me know if there's anything I can do to help upgrade Eqxvision. This is one of the best model zoos I know of. (And I'd like to try and see it get some more attention somehow.)
